examples/vision/3D_image_classification.py

```diff
@@ -0,0 +1,469 @@
+    # normalize
+    volume = tf_normalize(volume)
+    return volume, label
+
+
+def test_augment(volume, label):
+    # normalize
+    volume = tf_normalize(volume)
+    return volume, label
+
+
+"""
+While defining the train and test data loader, the training data is passed through and
+augmentation function which randomly rotates or blurs the volume and finally rescales it
+to have values between 0 to 1. 
+
+For the test data, the volumes are only rescaled.
+"""
+
+# define data loaders
+train_dataset = tf.data.Dataset.from_tensor_slices((x_train, y_train))
+test_dataset = tf.data.Dataset.from_tensor_slices((x_test, y_test))
+
+batch_size = 2
+# augment the on the fly during training
+train_dataset = train_dataset.shuffle(len(x_train)).map(train_augment).batch(batch_size)
+# only rescale
+test_dataset = test_dataset.shuffle(len(x_test)).map(test_augment).batch(batch_size)
+
+# data = train_dataset.take(1)
+# print(data)
+# images, labels = list(data)[0]
+# images.shape, labels.shape
+
+"""
+Visualize an augmented CT scan
+"""
+
+import matplotlib.pyplot as plt
+
+data = train_dataset.take(1)
+images, labels = list(data)[0]
+images = images.numpy()
+image = images[0]
+print("Dimension of the CT scan is:", image.shape)
+plt.imshow(np.squeeze(image[:, :, 80]), cmap="gray")
+
+# visualize montage of slices
+# 10 rows and 10 columns for 100 slices of the CT scan
+plot_slices(10, 10, 128, 128, image[:, :, :100])
+
+"""
+## Define 3D convolutional neural network
+
+
+
+To make the model easier to understand, blocks are defined. Since this is a
+3D CNN, 3D convolutions are used. The architecture of the 3D CNN used in this example
+is based on this [paper](https://arxiv.org/abs/2007.13224).
+"""
+
+
+def get_model(img_size, num_classes):
+    # build a 3D convolutional neural network model
+
+    inputs = keras.Input(img_size)
+
+    x = layers.Conv3D(filters=16, kernel_size=3, activation="relu")(inputs)
+    x = layers.MaxPool3D(pool_size=2)(x)
+    x = layers.BatchNormalization()(x)
+
+    x = layers.Conv3D(filters=32, kernel_size=3, activation="relu")(x)
+    x = layers.MaxPool3D(pool_size=2)(x)
+    x = layers.BatchNormalization()(x)
+
+    x = layers.Conv3D(filters=64, kernel_size=3, activation="relu")(x)
+    x = layers.MaxPool3D(pool_size=2)(x)
+    x = layers.BatchNormalization()(x)
+
+    x = layers.Conv3D(filters=128, kernel_size=3, activation="relu")(x)
+    x = layers.MaxPool3D(pool_size=2)(x)
+    x = layers.BatchNormalization()(x)
+
+    x = layers.Flatten()(x)
+    x = layers.Dense(units=256, activation="relu")(x)
+    x = layers.Dropout(0.3)(x)
+
+    outputs = layers.Dense(units=1, activation="sigmoid")(x)
+
+    # define the model
+    model = keras.Model(inputs, outputs, name="3dcnn")
+    return model
+
+
+# build model
+model = get_model((128, 128, 128, 1), 2)
+model.summary()
+
+"""
+## Train model
+"""
+
+# compile model
+initial_learning_rate = 0.0001
+lr_schedule = tf.keras.optimizers.schedules.ExponentialDecay(
```

Comment:
Since you import `keras`, you can replace `tf.keras` with `keras` throughout this file (saves 3 char)